core/adapters/eth_tx.go

```diff
@@ -25,12 +26,13 @@ const (
 // EthTx holds the Address to send the result to and the FunctionSelector
 // to execute.
 type EthTx struct {
-	Address          common.Address       `json:"address"`
+	ToAddress        common.Address       `json:"address"`
+	FromAddress      *common.Address      `json:"fromAddress"`
 	FunctionSelector eth.FunctionSelector `json:"functionSelector"`
 	DataPrefix       hexutil.Bytes        `json:"dataPrefix"`
 	DataFormat       string               `json:"format"`
 	GasPrice         *utils.Big           `json:"gasPrice" gorm:"type:numeric"`
```

Comment:
Does this need to be null? Pointer struct members actually make memory usage go up, so I wouldn't use them unless you really need null and it needs to be distinct from zero.

Reply:
I figured pointers were used when something could be omitted. It's unambiguous to check for `nil`.

You can legitimately have 0-priced transactions, they sometimes crop up on Ropsten.

Reply:
Re: `nil`-able fields --
Benefit: increase in clarity
Cost: node panics (or at least ends up in a weird state, if it's a sub-goroutine that panics) when a single task run has a bad input

Personally, I don't care about memory in cases like these, but I think the Go philosophy of preferring zero values over nil-able values is probably healthier for the software, as long as we have good error reporting.

Reply:
@j16r @spooktheducks this reasoning makes sense to me, will convert to use zero values instead of nils